### ISSUE TITLE
flycheck-clang-language-standard not useful

### ISSUE THREAD

**Issue by bo0ts** [flycheck/flycheck#333]
flycheck-clang-language-standard is supposed to be set to the language standard that should be used when checking C or C++ code (e.g. `c++11` or `c11`).

Unfortunately clang accepts different flags depending on whether C or C++ code is checked and will produce an error.

To reproduce:

```
(setq flycheck-clang-language-standard "c++11")
; find-file "/tmp/bla.c"
; enable flycheck-mode
```

Error message:

> Checker c/c++-clang returned non-zero exit code 1, but no errors from output: error: invalid argument '-std=c++11' not allowed with 'C/ObjC'

It might be necessary to provide two variables for C++ and C respectively.

**Comment by wudixiaotie** [comment]
@lunaryorn I added the hook like you said, but still got some warning like this

> range-based for loop is a C++11 extension (c/c++-clang)

here is the code:
```c++
#include <iostream>

using namespace std;

int main() {
  int a[4] = {1, 2, 3, 4};
  for (int s : a) {
    cout << s << endl;
  }
  return 0;
}
```
